apps/rush-lib/src/logic/taskRunner/TaskRunner.ts

```diff
@@ -194,47 +211,75 @@ export class TaskRunner {
    * Pulls the next task with no dependencies off the build queue
    * Removes any non-ready tasks from the build queue (this should only be blocked tasks)
    */
-  private _getNextTask(): Task | undefined {
-    for (let i: number = 0; i < this._buildQueue.length; i++) {
-      const task: Task = this._buildQueue[i];
+  private _getNextTaskSync(): IteratorResult<Task | undefined> {
```

Comment:
The name implies that this returns one task object.

Reply:
Refactored this into a separate class that handles async task queue iteration.